### ISSUE TITLE
Limited profiles can't be detected/switched

### ISSUE THREAD

**Issue by Darren-Hill** [mammuth/ha-fritzbox-tools#121]
On my Fritzbox, the kids profiles have time limited access to the internet (disallowed between 9pm and midnight, allowed at all other times).

If I try and set up switches on those profiles, they almost immediately become unavailable and basically don't work. It's due to the limitations set within the Fritzbox, as if I set them to unlimited internet access then they immediately become available again and switch on/off fine.

Is it possible to allow the limited profiles to also be switched via HA, or is it strictly one or the other?

Aside from that this is a very nice addition to HA with Fritzbox - many thanks for your work on it!

**Comment by AaronDavidSchneider** [comment]
Hi,

thanks for testing! 
The idea is that you replace the Fritzbox time automations by home assistant automations. Fritzbox Tools will only toggle between unlimited an never.

However, it should be very straightforward to have homeassistant toggle the profile switches for you.

**Comment by Darren-Hill** [comment]
OK, that's kinda what I figured (and why I asked if it's one or the other).

Would just be nicer to have both be able to control things, as the Fritzbox scheduler is quite nice to use (to be able to set different times per day, in a nice tabular input format). As you say I can do it via a HA schedule, but it gets more complex to do different stuff on different days. Was more hoping to just use HA as an "override" switch to manually turn on/off in parallel with the existing schedules.

**Comment by AaronDavidSchneider** [comment]
Ok. That makes sense. I will give it a thought. I am not sure if it's possible though.

Meanwhile you could use:
https://www.home-assistant.io/blog/2020/10/07/release-116/#automation-editor-has-now-weekday-support

And set it up that way.

**Comment by Darren-Hill** [comment]
Unfortunately no idea - it's a bit too in-deep for my skills.

I'm looking at what options are available, also [Shedy](https://hass-apps.readthedocs.io/en/stable/apps/schedy/index.html) but I'm not sure if I can use that on my set-up (I'm running HA under DietPi).

Thanks for the consideration on it though. I can work around with what we have available, so it's only a "nice to have" if you can see a way for it without too much work or effort.

**Comment by AaronDavidSchneider** [comment]
Yes, I think something like `input_select` would be the way to go. However, these are only frontend options while `fritzbox_tools` operates beyond that.

Alternatively, we could always toggle between `limited` and `never`. Or between `unlimited` and `limited`. 
What do you think?

I will ask in the home assistant community if there is the possibility for a input_select like tri state switch.

**Comment by Darren-Hill** [comment]
Of the choices, the toggle between limited and never would be best. Of course the issue will be for such toggling to know what to toggle back to, whether it's between never/limited or never/unlimited depending on which profile you're talking about (as I presume once you've toggled to never, you won't know where you came from to know where to go back to).

There are a few examples of requests for tri-state switches in the HA forum threads - I did a quick search before when I posted last time. Most of the recommendations are for input select, as there currently is no specific tri-state switch option.

It's a shame that there isn't more exposed in the API, as blocking at a device level is available under the parental controls. But I'm assuming neither that nor indeed the device list itself is exposed and available to HA.

**Comment by AaronDavidSchneider** [comment]
> Of the choices, the toggle between limited and never would be best. Of course the issue will be for such toggling to know what to toggle back to, whether it's between never/limited or never/unlimited depending on which profile you're talking about (as I presume once you've toggled to never, you won't know where you came from to know where to go back to).
> 
Yes thats indeed a problem. That means we will not be able to resolve this issue with a fix, since its the same problem if you would manually toggle to never or unlimited in the fritzbox. 

The user will need to setup automations in HA. This also seems to be the best way in my opinion.

> There are a few examples of requests for tri-state switches in the HA forum threads - I did a quick search before when I posted last time. Most of the recommendations are for input select, as there currently is no specific tri-state switch option.
> 
Yes but those are frontend options that are not applicable to `fritzbox_tools`

Thanks again for reporting and for your thoughts! I will close this issue now.
